### PR TITLE
A pom to be able to publish XSpec as a Maven Artifact

### PR DESCRIPTION
**Pull request by cmarchand**
Be careful, version information is in pom.xml, and this should be consistent with current way of creating releases on GitHub site.

To publish a release, you must
 * be identified with an account on oss.sonatype.org that can publish go io.xspec
 * have a valid GPG key to sign artifacts
 * run `mvn -P release deploy`

### CODE REVIEW

**Comment by cirulls** [comment]
@cmarchand : Thanks for this. Few initial comments from me:

- It would be good to add some documentation in the wiki on how to publish XSpec as a maven artifact. This could be useful as internal documentation for the next versions. 
- Regarding the version numbers, you used `0.6-SNAPSHOT` in the `pom.xml` for the current XSpec version genereated from `master`. Following discussions in #139, it looks like XSpec users favour a switch to v1.x.x for the next XSpec release. It would be good to align the version number in the `pom.xml` file.
- In the `developers` section, I think there should be your name as you developed this. It's fine to include my name and email but I am a bit concerned about adding Florent and Jeni without their explicit consent (they may not want to have their email listed here). I suggest to remove them for now, we can always include them later after they gave their consent.

**Comment by cmarchand** [comment]
Ok, I'm gonna do this.
I don't know if I can make push requests on Wiki...

What will be the next release ?

Best,
Christophe

December 20, 2017 2:05:55 PM CET Sandro Cirulli <notifications@github.com> wrote:
@cmarchand : Thanks for this. Few initial comments from me:It would be good to add some documentation in the wiki on how to publish XSpec as a maven artifact. This could be useful as internal documentation for the next versions.Regarding the version numbers, you used 0.6-SNAPSHOT in the pom.xml for the current XSpec version genereated from master. Following discussions in #139, it looks like XSpec users favour a switch to v1.x.x for the next XSpec release. It would be good to align the version number in the pom.xml file.In the developers section, I think there should be your name as you developed this. It's fine to include my name and email but I am a bit concerned about adding Florent and Jeni without their explicit consent (they may not want to have their email listed here). I suggest to remove them for now, we can always include them later after they gave their consent.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/xspec/xspec","title":"xspec/xspec","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/xspec/xspec"}},"updates":{"snippets":[{"icon":"PERSON","message":"@cirulls in #156: @cmarchand : Thanks for this. Few initial comments from me:\r\n\r\n- It would be good to add some documentation in the wiki on how to publish XSpec as a maven artifact. This could be useful as internal documentation for the next versions. \r\n- Regarding the version numbers, you used `0.6-SNAPSHOT` in the `pom.xml` for the current XSpec version genereated from `master`. Following discussions in #139, it looks like XSpec users favour a switch to v1.x.x for the next XSpec release. It would be good to align the version number in the `pom.xml` file.\r\n- In the `developers` section, I think there should be your name as you developed this. It's fine to include my name and email but I am a bit concerned about adding Florent and Jeni without their explicit consent (they may not want to have their email listed here). I suggest to remove them for now, we can always include them later after they gave their consent. "}],"action":{"name":"View Pull Request","url":"https://github.com/xspec/xspec/pull/156#issuecomment-353058145"}}}

**Comment by cirulls** [comment]
Looking at the feedback in #139, I'm leaning towards v1.0.0 as the next XSpec version number. 

Regarding the wiki, unfortunately you can't make pull request on it but if you send me a Markdown or a text file with the documentation I'll add it in the relevant section. Or I can give you write permission on the wiki, as you prefer...

**Comment by cirulls** [comment]
I removed this pull request for the milestone v1.0.0 as we still need to test it and add documentation. Hopefully it will be included in next release.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      <distribution>repo | manually</distribution>
+      <url>https://opensource.org/licenses/MIT</url>
+      <comments>Copyright (c) 2008-2017 Jeni Tennison</comments>
+    </license>
+  </licenses>
+  
+  <developers>
+    <developer>
+      <name>Jeni Tennison</name>
+      <email>jeni.@jenitennison.com</email>
```
Comment (cirulls) [inline pom.xml]: I think there is a typo, Jeni provided this email and gave her consent:

```
jeni@jenitennison.com
```

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    <developer>
+      <name>Sandro Cirulli</name>
+      <email>sandro.cirulli@oup.com</email>
+      <roles>
+        <role>current maintainer</role>
+      </roles>
+    </developer>
+    <developer>
+      <name>Florent George</name>
+      <email>info@fgeorges.org</email>
```
Comment (cirulls) [inline pom.xml]: Florent gave his consent but asked to use this email address:

```
 fgeorges@fgeorges.org
```

Comment (cmarchand) [reply pom.xml]: You are right, pull request is not up to date

**Comment by cmarchand** [comment]
Documentation has been published here : [http://blog.oxiane.com/2018/02/26/deploying-open-source-artifact-to-central-maven/](http://blog.oxiane.com/2018/02/26/deploying-open-source-artifact-to-central-maven/) and here : [https://github.com/cmarchand/cp-protocol/wiki/Deploying-Open-Source-artifact-to-Central-Maven](https://github.com/cmarchand/cp-protocol/wiki/Deploying-Open-Source-artifact-to-Central-Maven).

Password for io.xspec groupId is available from @cirulls and @cmarchand, but won't be published here.

When does this pull-request will be merged ?

**Comment by cirulls** [comment]
@cmarchand: sorry for the wait, unfortunately I've been very busy after XML Prague and couldn't do much work for XSpec. I'll review your pull request by the end of this week, sorry for keep you waiting.

**Comment by cirulls** [comment]
@cmarchand : thanks for this and sorry if it took me ages to review it. I put your documentation on the [wiki](https://github.com/xspec/xspec/wiki/Deploy-XSpec-artifact-to-Maven-Central). I slightly rephrased few sentences, I hope it's fine. Next time we do a release I'd like to try the process myself so that there are at least two people who can deploy to Maven repositories.

**Pull request merged**